filters/IterativeClosestPoint.cpp

```diff
@@ -281,15 +280,24 @@ PointViewPtr IterativeClosestPoint::icp(PointViewPtr fixed,
     // Compute the MSE one last time, using the unaltered, fixed PointView and
     // the transformed, moving PointView.
     double mse(0.0);
+    size_t mse_n(0);
+    double sqr_maxdist = m_maxdist * m_maxdist;
     KD3Index& kd_fixed_orig = fixed->build3dIndex();
     for (PointRef p : *moving)
     {
         PointIdList indices(1);
         std::vector<double> sqr_dists(1);
         kd_fixed_orig.knnSearch(p, 1, &indices, &sqr_dists);
+        // Check that the square distance does not exceed threshold value.
+        if (m_maxdistArg->set())
```

Comment:
Perhaps set sqr_maxdist to (std::numeric_limits<double>::max)() if m_maxdistArg() isn't set and then reverse the logic to avoid the `continue`:
```
if (sqr_dists[0] < sqr_maxdist)
{
  mse_n++;
  mse += std::sqrt(sqr_dists[0]);
}
```
This avoids the repeated test of the invariant above (don't *think* the compiler can figure that it's constant in this context).  Maybe this makes no real difference, though.

Reply:
I'm fine with that. Eliminating the nested `if` statements is cleaner. I pushed the change.